ext/zlib/zlib_fopen_wrapper.c

```diff
@@ -33,24 +33,55 @@ struct php_gz_stream_data_t	{
 static ssize_t php_gziop_read(php_stream *stream, char *buf, size_t count)
 {
 	struct php_gz_stream_data_t *self = (struct php_gz_stream_data_t *) stream->abstract;
-	int read;
+	ssize_t total_read = 0;
+
+	/* Despite the count argument of gzread() being "unsigned int",
+	 * the return value is "int". Error returns are values < 0, otherwise the count is returned.
+	 * To properly distinguish error values from success value, we therefore need to cap at INT_MAX.
+	 */
+	do {
```

Comment:
With `stream_wrapper_register` it should be possible to test large file.

- generate/stream data with some deterministic pattern like 1234...
- gzip encode them
- gzip decode them
- test if the read data matches the expected data/pattern

Reply:
That's true, but we would still need https://github.com/php/php-src/pull/14895 first to properly check for the memory limit